Java/LFU.java

```diff
@@ -0,0 +1,144 @@
+package lfu;
+import java.util.LinkedList;
+import java.util.ListIterator;
+import java.util.HashMap;
+
+/**
+ * @author salman
+ * @param <T>
+ */
+
+public class LFU<T> {
```

Comment:
Umm aren't long names not advised as a class? LFU is a bad name though. Coulldn't think of an alternative though!